lms/services/assignment.py

```diff
@@ -0,0 +1,67 @@
+from functools import lru_cache
+
+from lms.models import ModuleItemConfiguration
+
+
+class AssignmentService:
+    """A service for getting and setting assignments (ModuleItemConfiguration's)."""
+
+    def __init__(self, db):
+        self._db = db
+
+    def get_document_url(self, tool_consumer_instance_guid, resource_link_id):
+        """
+        Return the matching document URL or None.
+
+        Return the document URL for the assignment with the given
+        tool_consumer_instance_guid and resource_link_id, or None.
+        """
+        mic = self._get(tool_consumer_instance_guid, resource_link_id)
+
+        if mic:
+            return mic.document_url
```

Comment:
I don't know what the current convention is on the backend but my preference is always to have early-exits in a function be failure/no match cases (of which there may be many) and the success/match found path be the last return in the function

Reply:
Changed this to `return mic.document_url if mic else None`